util/collate/collate.go

```diff
@@ -18,16 +18,17 @@ import (
 )
 
 var (
-	collatorMap map[string]Collator
+	collatorMap   map[string]Collator
+	collatorIDMap map[int]Collator
 )
 
 // Collator provides functionality for comparing strings for a given
 // collation order.
 type Collator interface {
 	// Compare returns an integer comparing the two strings. The result will be 0 if a == b, -1 if a < b, and +1 if a > b.
-	Compare(a, b string) int
+	Compare(a, b string, padLen int) int
```

Comment:
There may be other arguments about padding like `padChar`, `ignorePad` and etc, so how about using a struct here which is easier to add some new arguments in the future.